include/GoogleSync/GoogleSync.php

```diff
@@ -495,107 +657,67 @@ public function setGService()
             return false;
         }
 
-        // If we only got one event, figure out which kind it is
+        // Did we only get one event?
         if (!isset($event_2) || empty($event_2)) {
-            switch (get_class($event_1)) {
-                case "Google_Service_Calendar_Event":
-                    if ($event_1->status !== 'cancelled') {
-                     // We only pull if the Google Event is not deleted/cancelled
-                        $return = "pull";
-                    } else {
-                        $return = "skip";
-                    }
-                    break;
-                case "Meeting":
-                    if ($event_1->deleted == '0') {
-                     // We only push if the meeting is not deleted
-                        $return = "push";
-                    } else {
-                        $return = "skip";
-                    }
-                    break;
-                default:
-                    $GLOBALS['log']->fatal(__FILE__ . ':' . __LINE__ . ' ' . __METHOD__ . ' - ' . 'I don\'t understand the event type you used.');
-                    $return = false;
-            }
-            // Only one event, we're done here
-            return $return;
+            // If we only got one event, figure out which kind it is, and pass the return from the helper function
+            return $this->singleEventAction($event_1);
         } else {
-
-            // We have two events... figure out which is which and set the internal objects
-            $obj_class1 = get_class($event_1);
-            if ($obj_class1 == 'Meeting') {
-                $event_local = $event_1;
-            } elseif ($obj_class1 == 'Google_Service_Calendar_Event') {
-                $event_remote = $event_1;
-            } else {
-                $GLOBALS['log']->fatal(__FILE__ . ':' . __LINE__ . ' ' . __METHOD__ . ' - ' . 'Events must be of type \'Meeting\' or \'Google_Service_Calendar_Event\'');
-                return false;
-            }
-
-            $obj_class2 = get_class($event_2);
-            if ($obj_class2 == 'Meeting') {
-                $event_local = $event_2;
-            } elseif ($obj_class2 == 'Google_Service_Calendar_Event') {
-                $event_remote = $event_2;
-            } else {
-                $GLOBALS['log']->fatal(__FILE__ . ':' . __LINE__ . ' ' . __METHOD__ . ' - ' . 'Events must be of type \'Meeting\' or \'Google_Service_Calendar_Event\'');
-                return false;
-            }
-
-            if (in_array($event_local->id, $this->syncedList, true)) {
-                $GLOBALS['log']->info(__FILE__ . ':' . __LINE__ . ' ' . __METHOD__ . ' - ' . 'We already synced this meeting. Marking to skip.');
-                return "skip";
-            }
+            // For two events, pass to the helper function that sorts them and returns an array of objects
+            $events_array = $this->sortEventObjects($event_1, $event_2);
```

Comment:
sortEventObjects() return value could be a boolean but used as an array as in_array() parameter few line above. Please, handle the boolean return value.

Reply:
Done. Throws exception if it returns false. Should never happen.